test/src/app/App.test.js

```diff
@@ -145,7 +147,8 @@ contract('App', function ([_, owner]) {
 
       const shouldUpgradeProxy = function () {
         it('should upgrade proxy to ImplV2', async function () {
-          (await this.proxy.version()).should.eq('V2');
+          await this.proxy.version().should.eventually.be.eq('V2');
+          await this.app.getProxyImplementation(this.proxy.address).should.eventually.be.eq(this.implementation.address)
```

Comment:
I think this test would be better if `this.implementation` was instead `this.implementation_2`, or something like that.

`this.implementation` is used both for the first implementation and the second one. That sounds a bit fishy.